all-actors/src/main/java/org/sunbird/cert/actor/TemplateValidateActor.java

```diff
@@ -0,0 +1,106 @@
+package org.sunbird.cert.actor;
+
+
+import org.apache.log4j.Logger;
+import org.incredible.certProcessor.JsonKey;
+import org.incredible.certProcessor.store.LocalStore;
+import org.incredible.certProcessor.views.HTMLTemplateValidator;
+import org.incredible.certProcessor.views.HTMLTemplateZip;
+import org.sunbird.BaseActor;
+import org.sunbird.BaseException;
+import org.sunbird.CertsConstant;
+import org.sunbird.actor.core.ActorConfig;
+import org.sunbird.cloud.storage.exception.StorageServiceException;
+import org.sunbird.message.IResponseMessage;
+import org.sunbird.message.ResponseCode;
+import org.sunbird.request.Request;
+import org.sunbird.response.Response;
+
+import java.io.IOException;
+import java.text.MessageFormat;
+import java.util.Set;
+
+/**
+ * @author Aishwarya
+ * This actor is responsible for Template validation
+ */
+@ActorConfig(
+        tasks = {JsonKey.VALIDATE_TEMPLATE},
+        asyncTasks = {}
+)
+public class TemplateValidateActor extends BaseActor {
+
+    private Logger logger = Logger.getLogger(TemplateValidateActor.class);
+    private CertsConstant certsConstant = new CertsConstant();
+
+    @Override
+    public void onReceive(Request request) throws Throwable {
+        String operation = request.getOperation();
+        logger.info("onReceive method call start for operation " + operation);
+        if (JsonKey.VALIDATE_TEMPLATE.equalsIgnoreCase(operation)) {
+            validateTemplate(request);
+        }
+    }
+
+
+    private void validateTemplate(Request request) throws BaseException {
+        HTMLValidatorResponse validatorResponse;
+        String templateUrl = (String) request.getRequest().get(JsonKey.TEMPLATE_URL);
+        HTMLTemplateZip htmlTemplateZip = new HTMLTemplateZip(new LocalStore(certsConstant.getDOMAIN_URL()), templateUrl);
+
+        //download the file
+        try {
```

Comment:
Please try to write less code. Why can't this be a single try-multiple catch blocks - I see invalid zip file or error downloading are the only 2 kinds of errors.
Am finding cleanup line, throw repeated.

Reply:
done